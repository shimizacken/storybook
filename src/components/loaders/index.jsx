import React from 'react';
import Hearts from 'react-svg-loader!../../assets/images/svg/hearts.svg';
import ThreeDots from 'react-svg-loader!../../assets/images/svg/three-dots.svg';
import Grid from 'react-svg-loader!../../assets/images/svg/grid.svg';

const HeartsLoader = (width, height) => (
    <Hearts width={width} height={height} />
);

const ThreeDotsLoader = (width, height) => (
    // <ThreeDots width={width} height={height} />
    <svg width='35' viewBox='0 0 120 30'
        fill='#000'>
    <circle cx='15' cy='15' r='15'>
        <animate attributeName='r' from='15' to='15'
                 begin='0s' dur='0.8s'
                 values='15;9;15' calcMode='linear'
                 repeatCount='indefinite' />
        <animate attributeName='fillOpacity' from='1' to='1'
                 begin='0s' dur='0.8s'
                 values='1;.5;1' calcMode='linear'
                 repeatCount='indefinite' />
    </circle>
    <circle cx='60' cy='15' r='9' fillOpacity='0.3'>
        <animate attributeName='r' from='9' to='9'
                 begin='0s' dur='0.8s'
                 values='9;15;9' calcMode='linear'
                 repeatCount='indefinite' />
        <animate attributeName='fillOpacity' from='0.5' to='0.5'
                 begin='0s' dur='0.8s'
                 values='.5;1;.5' calcMode='linear'
                 repeatCount='indefinite' />
    </circle>
    <circle cx='105' cy='15' r='15'>
        <animate attributeName='r' from='15' to='15'
                 begin='0s' dur='0.8s'
                 values='15;9;15' calcMode='linear'
                 repeatCount='indefinite' />
        <animate attributeName='fillOpacity' from='1' to='1'
                 begin='0s' dur='0.8s'
                 values='1;.5;1' calcMode='linear'
                 repeatCount='indefinite' />
    </circle>
</svg>
);

const GridLoader = (width, height) => (
    <Grid width={width} height={height} />
);

export {
    HeartsLoader,
    ThreeDotsLoader,
    Grid
};