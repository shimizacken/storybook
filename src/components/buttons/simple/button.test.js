import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './index';
import { ThreeDotsLoader } from '../../loaders';

describe('simple button test suite', () => {

    describe('button text', () => {


        it('should render', () => {

            const wrapper = shallow(<Button />);
    
            expect(wrapper).toMatchSnapshot();
        });
    
        it('should render text', () => {
    
            const text = 'Save';
    
            const wrapper = shallow(<Button>{text}</Button>);
            
            expect(wrapper).toMatchSnapshot();
        });
    
        it('should update text', () => {
    
            const text = 'Save';
    
            const wrapper = shallow(<Button>{text}</Button>);
            
            //console.log(wrapper.html());
    
            wrapper.setProps({
                children: 'Continue'
            });
    
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('button attributes', () => {

        it('render id attribute', () => {

            const text = 'Save', id = 'btnSave';

            const wrapper = shallow(<Button id={id}>{text}</Button>);
            
            //expect(wrapper.find('button').prop('id')).toBe(id);
            expect(wrapper).toMatchSnapshot();
        });

        it('render data-cy attribute', () => {

            const text = 'Save', id = 'btnSave', dataCy = 'data-cy';

            const wrapper = shallow(<Button 
                                        id={id}
                                        dataCy={dataCy}
                                    >
                                        {text}
                                    </Button>);
            
            //expect(wrapper.find('button').prop('data-cy')).toBe(dataCy);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('states of the button', () => {

        it('render disabled', () => {

            const text = 'Save', id = 'btnSave', dataCy = 'data-cy';

            const wrapper = shallow(<Button 
                                        id={id}
                                        dataCy={dataCy}
                                        disabled
                                    >
                                        {text}
                                    </Button>);
            
            //expect(wrapper.hasClass('disabled')).toBeTruthy();
            expect(wrapper).toMatchSnapshot();
        });

        it('render inProgress', () => {

            const text = 'Save', id = 'btnSave', dataCy = 'data-cy';

            const wrapper = shallow(<Button 
                                        id={id}
                                        dataCy={dataCy}
                                        inProgress
                                    >
                                        {text}
                                    </Button>);
            
            //console.log(wrapper.text());
            //expect(wrapper.hasClass('disabled')).toBeTruthy();
            //expect(wrapper.find(ThreeDotsLoader).exists()).toBeTruthy();
            expect(wrapper).toMatchSnapshot();
        });
    });
});