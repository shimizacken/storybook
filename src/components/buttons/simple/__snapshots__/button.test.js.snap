// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple button test suite button attributes render data-cy attribute 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    dataCy="data-cy"
    id="btnSave"
  >
    Save
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Save",
      "className": "simpleButton",
      "data-cy": "data-cy",
      "id": "btnSave",
      "onClick": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": "Save",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Save",
        "className": "simpleButton",
        "data-cy": "data-cy",
        "id": "btnSave",
        "onClick": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": "Save",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`simple button test suite button attributes render id attribute 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    id="btnSave"
  >
    Save
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Save",
      "className": "simpleButton",
      "data-cy": undefined,
      "id": "btnSave",
      "onClick": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": "Save",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Save",
        "className": "simpleButton",
        "data-cy": undefined,
        "id": "btnSave",
        "onClick": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": "Save",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`simple button test suite button text should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "simpleButton",
      "data-cy": undefined,
      "id": undefined,
      "onClick": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "simpleButton",
        "data-cy": undefined,
        "id": undefined,
        "onClick": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`simple button test suite button text should render text 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button>
    Save
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Save",
      "className": "simpleButton",
      "data-cy": undefined,
      "id": undefined,
      "onClick": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": "Save",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Save",
        "className": "simpleButton",
        "data-cy": undefined,
        "id": undefined,
        "onClick": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": "Save",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`simple button test suite button text should update text 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button>
    Continue
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Continue",
      "className": "simpleButton",
      "data-cy": undefined,
      "id": undefined,
      "onClick": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": "Continue",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Continue",
        "className": "simpleButton",
        "data-cy": undefined,
        "id": undefined,
        "onClick": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": "Continue",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`simple button test suite states of the button render disabled 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    dataCy="data-cy"
    disabled={true}
    id="btnSave"
  >
    Save
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Save",
      "className": "simpleButton disabled",
      "data-cy": "data-cy",
      "id": "btnSave",
      "onClick": [Function],
      "style": undefined,
    },
    "ref": null,
    "rendered": "Save",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Save",
        "className": "simpleButton disabled",
        "data-cy": "data-cy",
        "id": "btnSave",
        "onClick": [Function],
        "style": undefined,
      },
      "ref": null,
      "rendered": "Save",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`simple button test suite states of the button render inProgress 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    dataCy="data-cy"
    id="btnSave"
    inProgress={true}
  >
    Save
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ThreeDotsLoader
        height="15px"
        width="15px"
      />,
      "className": "simpleButton disabled",
      "data-cy": "data-cy",
      "id": "btnSave",
      "onClick": [Function],
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "height": "15px",
        "width": "15px",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ThreeDotsLoader
          height="15px"
          width="15px"
        />,
        "className": "simpleButton disabled",
        "data-cy": "data-cy",
        "id": "btnSave",
        "onClick": [Function],
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "height": "15px",
          "width": "15px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
